
syntax = "proto2";
option java_package = "cn.techwolf.boss.chat";
option java_outer_classname = "ChatProtocol";
//??¨??·è?????
message TechwolfUser {
    //??¨??·uid
    required int64 uid = 1;
    //??¨??·?§????,?????????è????¨è????¨,???è??????????????,???è??????????????
    optional string name = 2;
    //??¨??·?¤????,????????????è??è?????è??????????????è??è????????
    optional string avatar = 3;
    //??????
    optional string company = 4;
    //é??è?¤?¤????id
    optional int32 headImg = 5;
    //bossè?¤è????????
    optional int32 certification = 6;
}

//è??é??
message TechwolfSound {
    //????±?ID ???????????¨
    optional int64 sid = 1;
    //è??é????????url
    optional string url = 2;
    //?????????é??
    optional int32 duration = 3;
    //??·???  NULL|0|1:???è??é?? 2:è??é???????????
    optional int32 templateId = 4;                                      // ADD SINCE V6.13
}

//è??é??
message TechwolfVideo {
    // 1:è??é?? 2:è§?é??
    required int32 type = 1;

    // è???¤??????? 1:??????é??è?? 2:é??è???????? 3:???????????? 4:é??è??è?????
    required int32 status = 2;

    //é??è?????é??
    optional int32 duration = 3;
}

//??????
message TechwolfInterview {

    //1:???è?·é??è??è?·?±? 2:??????é??è??è?·?±? 3:??????é??è??è?·?±? 4:??????é??è??è?·?±? 5:??????é??è?? 6:???é??è??????¤?
    required int32 condition = 1;

    required string text = 2;

    optional string url = 3;

    optional string extend = 4;
}

//????????????
message TechwolfImageInfo {
    //??????url
    required string url = 1;
    //???
    required int32 width = 2;
    //é??
    required int32 height = 3;
}

//????????????
message TechwolfImage {
    //????±?id,???????????¨
    optional int64 iid = 1;
    //?????????
    optional TechwolfImageInfo tinyImage = 2;
    //????§???????
    optional TechwolfImageInfo originImage = 3;
}

//??¨???
message TechwolfAction {
    //??¨???id  0???é????? 1??????é??????¤?2????????????è??3???????????????  4???è?·?±?é??è??5?????????é??è??6?????????é??è??7?????¤???è???????????8???????????¤???9???????????¤??? -- è?????è??3.0????§???????
    //10???????????????è??è???????????è??3
    //11 ???????????????é?? 12 ???è?????????????° 13 ???è??é????? ????±??±???§???{"phone":"13811111111"}
    //14 ???è??é??è??é?????
    //15 ?????????é???????? {"text":"?????????é??????????¨??????????¤?", "request_id":123,  "job_id":123, "create_time":123456789, "server_time":123456789}
    //16 boss???è????? {"text":"boss?·????????????¨?????????", "request_id":123, "create_time":123456789, "server_time":123456789}
    //17 bossé??è??????????? {"text":"bossé??è??????????????????????????????§?è???¤????"}
    //18 boss??????????????? {"text":"???é????????boss?????????????????????"}
    //19 boss???????????????????§?è???¤?
    //20 ???è???????????action {"msg_id":1234, "action":1} action????????????????????????é??
    //21 è?·?±??????? 22 ????????????è?·?±? 23 ????????????è?·?±?
    //24???è?·?±?é??è?? 25?????????é??è??26?????????é??è??27?????¤???è???????????28???????????¤???29???????????¤???
    //30????????????è????? {"count":10, "info":{"title":"xxx", "desc":"xxx", timeMills:11111}} 31?????????????????????è?? {"count":10, "info":{"title":"xxx", "desc":"xxx", timeMills:11111}} 35????????°????????°è?????{"count":10}
    //32???è?·?±???¤????????? {"weixin":"leonlau"}
    //33???????????¤????????? {"weixin":"leonlau"}
    //34???????????¤?????????
    //35???è§????é??
    //36??????é?????é?????é?????é??é????¨???
    //37 è?·?±?è?·????????????38è?·??????????????? 39è?·??????????????????{"resumeEmail":"gy13zyl@163.com"}
    //40???è?·?±????é???????? 41??????é?????????????? 42??????é??????????????
    //43 ???????????¨è????????{"newCount":"10","totalCount":"15"}
    //44 ??????è?????action
    //45 ??°????????????
    //46 ??????è????·?????????
    //47 ????????????é??è??é??è?·
    //48 ????????????é??è??é??è?·
    //49 é??è??é??è?·???????¤?
    //50 ???é??è???????°????¤?
    //51 ?·????????????????é?????????????????id {"expectId": 12345}
    //53 ??????è§?é??è???¤????token??°c??? {"accessToken":"xxxxxxx"}
    //54 ???????????????????????? {"sourceTitle":"xxxxxxx"}
    //55 Link?????¨é???????? {"appLink":"bosszp://xxxx"}
    //56 ?????????é??è??
    //57 APP??????Web???é??
    //61 boss????????????è?·é??è??é??è?·
    //62 boss??????é??è??é??è?·
    //63 boss???é??è??é??è?·è????????
    //64 ??????/boss??????é??è??
    //65 ????°?é??è??
    //66 é??è?????é????°
    //67 é??è??è?????
    //68 ??????/boss?¤±???
    //69 é??è??é??è?·?????????
    //70 ?????¨
    //71 ??????è§??????????é?????è§?é????????é???????? {"secretKey":"base64EncodedKey"}
    required int32 aid = 1;
    //????±???°???
    optional string extend = 2;
}

//????????????
message TechwolfArticle {
    //???é??
    required string title = 1;
    //???è?°
    required string description = 2;
    //????????°???
    required string picUrl = 3;
    //???è?????è??url
    required string url = 4;
    //??·???: 0.?¤§????¨???? 1.?°?????¨???? 2.??¨?±??¨???? 3.?????°???????¨????@SinceV5.48 4.1802?¤§?????? 5.1802?°???????
    //      6.1804??¤??????é?????
    optional int32 templateId = 5;
    //???é?¨???é????????
    optional string bottomText = 6;
    //è????????é?? ms
    optional int64 timeout = 7;
    //???è???????° {"p2":"", "p3":""...,"p8":""}
    optional string statisticParameters = 8;
    //é?????é?¨???????¨????é?¨???2?¤???¨???
    repeated TechwolfSlice highlightParts = 9;
    //?¨????é?¨???
    repeated TechwolfSlice dimParts = 10;
    //??????é??
    optional string subTitle = 11;
}

//é?????
message TechwolfNotify {
    //??????
    required string text = 1;
    //???è?????è??url
    optional string url = 2;
}

//???é??
message TechwolfButton {
    //???é????????
    required string text = 1;
    //???è?????è??url
    optional string url = 2;
    //???é???¨????
    optional int32 templateId = 3;
}

//???è?????
message TechwolfDialog {
    //????¤???????
    required string text = 1;
    //???é??
    repeated TechwolfButton buttons = 2;
    //???????·????è????????
    required bool operated = 3;
    //?????????è??????¤????
    optional bool clickMore = 4;
    //???è??????±???? 0 ???é????¨?±???? 1????????????è????? 2????????????è????? 3?????¨?±?é????? 4??????é??è?? 5?????¨?±???°????????? 6?????????è??é???????·????¤? 7?????????è??è?????è????? 8?????????????????????è??????????????è?? 9???é??è??alert?????? 10???é??è??è?????è??????????????é???????? 11???è??????????????è?????
    optional int32 type = 5;
    //???è?????è????????
    optional string backgroundUrl = 6;
    //è????????é?? ms
    optional int64 timeout = 7;
    //???è???????° {"p2":"", "p3":""...,"p8":""}
    optional string statisticParameters = 8;
    //title type=4???é??è????¨ 5,??¨?±???°?????????
    optional string title = 9;
    //???è????? type=4
    optional string url = 10;
}

message TechwolfJobDesc {
    //???é??
    required string title = 1;
    //??????
    required string company = 2;
    //è???°?
    required string salary = 3;
    //???è?????è??
    required string url = 4;
    //è?????id
    required int64 jobId = 5;
    //è??????±????
    optional string positionCategory = 6;
    //???é??è???±?
    optional string experience = 7;
    //??????è???±?
    optional string education = 8;
    //??????
    optional string city = 9;
    //bossè?????
    optional string bossTitle = 10;
    //boss????????????
    optional TechwolfUser boss = 11;
    //???è??lid
    optional string lid = 12;
    //è??è??é?????
    optional string stage = 13;
    //???é?¨???????????¨??·è????????è?·???è?°
    optional string bottomText = 14;
    //è??????????¨
    optional string jobLabel = 15;
    //???????±??¤???????è????????è?¨?¤???? 1.???????????? 2.??????è§?é????? 4.è§?é??/???????±??¤?
    optional int32 iconFlag = 16;
}

message TechwolfResume {
    //??????????????????
    required TechwolfUser user = 1;
    //???è?°
    optional string description = 2;
    //????????????
    optional string city = 3;
    //??????è?????
    optional string position = 4;
    //??????
    repeated string keywords = 5;
    //??????id
    optional int64 expectId = 6;
    //???è??lid
    optional string lid = 7;
    //??§???
    optional int32 gender = 8;
    //??????è??è??
    optional string salary = 9;
    //?·???????é??
    optional string workYear = 10;
    //??????è????????
    optional string content1 = 11;
    //??????è????????
    optional string content2 = 12;
    //??????è???±?
    optional string education = 13;
    //???é??
    optional string age = 14;
    //??¨??·????????????
    repeated string labels = 15;
    //??¨??·????·????????????????è????????
    repeated UserExperience experiences = 16;
    //??????è??????±????
    optional string positionCategory = 17;
    //??????è?????è??è??
    optional string jobSalary = 18;
    //???é?¨???????????¨??·è????????è?·???è?°
    optional string bottomText = 19;
    //???è?·???????????¨??·?????????è??????????°??????é??
    optional string applyStatus = 20;
    //è?????id
    optional int64 jobId = 21;
}

message TechwolfHyperLink {
    required string text = 1;
    required string url = 2;
    //1,é??è§??????? 2, ??????é??è??é??è?·???3?????????é??è??, 4?????°?????????é?????, 5???é??è?????é?? 6. é????????7. è???¤?è?°???
    required int32 hyperLinkType = 3;
    //????±???????????·??????????KV??????
    //?????????????????? fileSizeDesc:???????¤§?°????è?° mailTip:é????????é?????é??
    optional string extraJson=4;
}

//????????????
message TechwolfMessageBody {
    //???????±???? 1.?????? 2.è??é?? 3.?????? 4.??¨??? 5.?????? 6.é????? 7.???è????? 8.è????? 9.?????? 10.??????
    //11.è?????è????? 12.è??é????? 13.è§?é???????? 14.???????????? 15.?¤??????? 16.????¤??????? 17.?????¨???é?? 18.è??????????? 19.???????????? 20.??¨???è?¨???
    //22.è???¤??????? 23.é??è????????
    required int32 type = 1;
    //???????¨????id 1.?????? 2.?±???? 3.?±??????°è?? 4.???è??????·§??·??? 5.è§?????????? 6.????????°è???±???? 7.é????? 8.è???????????è??é??
    required int32 templateId = 2;
    //?¤?é?¨??????
    optional string headTitle = 11;
    //???????????? ???type???1???,???é??è?????; è??é??è?????????????????è?????
    optional string text = 3;
    //è??é???????? ???type???2?????????é??è?????
    optional TechwolfSound sound = 4;
    //???????????? ???type???3?????????é??è?????
    optional TechwolfImage image = 5;
    //??¨????????? ???type???4?????????é??è?????
    optional TechwolfAction action = 6;
    //???????????? ???type???5?????????é??è?????
    repeated TechwolfArticle articles = 7;
    //é????? ???type???6?????????é??è?????
    optional TechwolfNotify notify = 8;
    //???è????? ???type???7?????????é??è?????
    optional TechwolfDialog dialog = 9;
    //è????????
    optional TechwolfJobDesc jobDesc = 10;
    //?????????
    optional TechwolfResume resume = 12;
    //?????????
    optional TechwolfRedEnvelope redEnvelope = 13;
    //è????????è?¨è?????
    optional TechwolfOrderDetail orderDetail = 14;
    //è??é?????
    optional TechwolfHyperLink hyperLink = 15;
    //è§?é??è???¤?
    optional TechwolfVideo video = 16;
    //????????????
    optional TechwolfInterview interview = 17;
    //è???????????
    optional TechwolfJobShare jobShare = 18;
    //????????????
    optional TechwolfResumeShare resumeShare = 19;
    //AT??????
    optional AtInfo atInfo = 20;
    //??¨???è?¨???
    optional TechwolfSticker sticker = 21;
    //è???¤???????
    optional TechwolfChatShare chatShare = 22;
    //é??è????????
    optional TechwolfInterviewShare interviewShare = 23;             // ADD SINCE V6.13
}

//??????
message TechwolfMessage {
    //?????????é??è??
    required TechwolfUser from = 1;
    //????????????è??
    required TechwolfUser to = 2;
    //???????±????: 1.???è?? 2.??¤è?? 3.?????? 4.????????????F2???é??è§? 5.??¤???????????? 6.??¤????????????F2???é??è§?
    required int32 type = 3;
    //??????id, ???è?????????????????è???????????
    optional int64 mid = 4;
    //???????????????é?? ???è?????????????????è????????????????¨?????°???é??
    optional int64 time = 5;
    //????????????
    required TechwolfMessageBody body = 6;
    //?????????????????????
    optional bool offline = 7;
    //??????è???????·?????????è??
    optional bool received = 8;
    //??¨é????????
    optional string pushText = 9;
    //??????id
    optional int64 taskId = 10;
    //?????·???msgid
    optional int64 cmid = 11;
    //???????????? 0 ??°?????? 1 ?·????è?° 2 ?·?è??
    optional int32 status = 12;
    //??????è????°?????? 0 é??è?¤???1???è????°
    optional int32 uncount = 13;
    //??¨é???????°é?? //1 é??è?¤??°é?????2???é?±?????°é??
    optional int32 pushSound = 14;
    //?????????è?°????????¨???2???è?¨?¤??????·?????¨??·???è????????0.?????? 1.BOSS 2.é????? 3.é?????
    optional int32 flag = 15;
    //??????????????????
    optional bytes encryptedBody = 16;
}

//?????·?????????
message TechwolfClientInfo {
    //?????·?????????
    optional string version = 1;
    //??????????????????
    optional string system = 2;
    //????????????????????????
    optional string systemVersion = 3;
    //??????
    optional string model = 4;
    //è???¤???·
    optional string uniqid = 5;
    //??????
    optional string network = 6;
    //appid
    optional int32 appid = 7;
    //platform
    optional string platform = 8;
    //???é??id
    optional string channel = 9;
    //ssid
    optional string ssid = 10;
    //mac
    optional string bssid = 11;
    //??¨??·?????????
    optional double longitude = 12;
    //??¨??·?????????
    optional double latitude = 13;
}


//?????·??????é????????
message TechwolfClientTime {
    //?????¨???é??
    optional int64 startTime = 1;
    //??¤é??
    optional int64 resumeTime = 2;
}

//????????????
message TechwolfPresence {
    //???????±????, 1.??????, 2.é??è????????è?????3.?????? 4. ?????°???Android??? 5. ????¤??????°???Android???
    required int32 type = 1;
    //?????????
    required int32 uid = 2;
    //?????·??????????????????????????????é??è??
    optional TechwolfClientInfo clientInfo = 3;
    //?????·??????????????????????????°???????¤??????°???é??è??
    optional TechwolfClientTime clientTime = 4;
    //???????????????lastMessageId
    optional int64 lastMessageId = 5;
    //???????????????lastGroupMessageId
    optional int64 lastGroupMessageId = 6;
}

//iq??????
message TechwolfKVEntry {
    required string key = 1;
    required string value = 2;
}

//iq?????????
message TechwolfIq {
    //??????id
    required int64 qid = 1;
    //è?·?±?è·????
    required string query = 2;
    //?????°
    repeated TechwolfKVEntry params = 3;
}

//????????????
message TechwolfIqResponse {
    //??????id
    required int64 qid = 1;
    //è?·?±?è·????
    required string query = 2;
    //è???????????
    repeated TechwolfKVEntry results = 3;
}

//??????id?????????è??
message TechwolfMessageSync {
    required int64 clientMid = 1;
    required int64 serverMid = 2;
}

//???????·?è??
message TechwolfMessageRead {
    required int64 userId = 1;
    required int64 messageId = 2;
    required int64 readTime = 3;
    optional bool sync = 4 [default = false];
}

//è???¤????è??
message TechwolfChatProtocol {
    //???è??type 1:message 2: presence 3: iq 4: iqResponse 5:sync 6:read 7:datasync
    required int32 type = 1;
    //???è???????????é??è?¤???1.0 ?????????
    optional string version = 2;
    //type???1?????????é??è?????
    repeated TechwolfMessage messages = 3;
    //type???2??????é??è?????
    optional TechwolfPresence presence = 4;
    //type???3?????????é??è?????
    optional TechwolfIq iq = 5;
    //type???4?????????é??è?????
    optional TechwolfIqResponse iqResponse = 6;
    //type???5?????????é??è?????
    repeated TechwolfMessageSync messageSync = 7;
    //type???6?????????é??è?????
    repeated TechwolfMessageRead messageRead = 8;

    //type???7?????????é??è???????°?????????
    optional TechwolfDataSync dataSync = 9;
    //????????? 0:????????? 1:???è?? 2:??¤è??
    optional int32 domain = 10;
}

message TechwolfRedEnvelope {
    //??????id
    required int64 redId = 1;
    //?????????è¨?
    required string redText = 2;
    //?????????é??
    required string redTitle = 3;
    //????????°???
    required string clickUrl = 4;

}

//è?????
message TechwolfOrderDetail {
    //è?????????§°
    required string title = 1;
    //??????é??
    required string subTitle = 2;
    //?????????é?????
    optional string url = 3;
    //????¤???????è?¨
    repeated TechwolfOrderDetailEntry orderDetailEntryList =  4;
}

//è?????????¤?é??
message TechwolfOrderDetailItem {
    //????¤????
    required string name = 1;//?????????key:___line___ é????°??¤key??????value
    //?¨?????±????
    required int32 templateId = 2;
}

//è????????entry
message TechwolfOrderDetailEntry {
    //key
    required TechwolfOrderDetailItem key = 1;
    //value
    required TechwolfOrderDetailItem value = 2;
}

//??¨??·????????°?????????
message TechwolfUserSync {

    // ??¨??·id
    required int64 uid = 1;
    // ????????¨??·uid???è?????
    required int32 identity = 2;

    //??????type 3???????????????JSON??°????????? F3/F4?????¨??????
    optional string extraJson = 3;
}


// ??°?????????
message TechwolfDataSync {

    //??°??????????±???? 1:?????°????????????,2:?????°????????????,3:F3/F4?????¨??????
    //2001:??¤???????????°, 2002:??¤???????????°, 2003:????????¨??·??¤??????é??????????°
    required int32 type = 1;
    //??¨??·??????,type: 1
    optional TechwolfUserSync userSync = 2;
    //??¤????????????
    optional TechwolfGroupSync groupSync = 3;
}


//??????
message TechwolfSlice {
    //è?·?§???????
    required int32 startIndex = 1;
    //???????????????????????????
    required int32 endIndex = 2;
}

//??¨??·??????,??¨???C/B??????????±??¤?
message UserExperience {
    //??????(???????????????)
    required string organization = 1;
    //é?????(è??????????????)
    required string occupation = 2;
    //????§????é??
    optional string startDate = 3;
    //?????????é??
    optional string endDate = 4;
    //?±?????±???????1????·?????????????2??????è????????
    required int32 type = 5;
}

//??????è?????
message TechwolfJobShare {
    //BOSS??????: uid,name,avatar,company,certification
    required TechwolfUser user = 1;
    //è?????id
    required int64 jobId = 2;
    //è????????
    required string position = 3;
    //è???°?
    required string salary = 4;
    //??°???
    optional string location = 5;
    //??????
    required string company = 6;
    //è??è??é?????
    optional string stage = 7;
    //???é??è???±?
    optional string experience = 8;
    //??????è???±?
    optional string education = 9;
    //è·?è?????è??é?????
    optional string url = 10;
    //???è??lid
    optional string lid = 11;
}

//????????????
message TechwolfResumeShare {
    //??????????????????: uid,name,avatar,company,certification
    required TechwolfUser user = 1;
    //??????id
    required int64 expectId = 2;
    //??????è?????
    required string position = 3;
    //??????è??è??
    required string salary = 4;
    //????????°???
    optional string location = 5;
    //???è?·???????????¨??·?????????è??????????°??????é??
    optional string applyStatus = 6;
    //???é??
    optional string age = 7;
    //?·?é??
    optional string experience = 8;
    //??????
    optional string education = 9;
    //è·?è?????è??é?????
    optional string url = 10;
    //???è??lid
    optional string lid = 11;
    //??§??????-1:é??è?¤, 0:???, 1:??·???
    optional int32 gender = 12;
    //???????¨?????¤????
    optional bool blurred = 13;
    //???????????????NULL|0|1:??¤è??, 2:è??????????????
    optional int32 source = 14;
}

//AT??????
message AtInfo {
    //AT???è?°: 0.??? 1.AT????????????è??(???è??)
    required int32 flag = 1;
    //AT??¨??·???è?¨
    repeated int64 uids = 2;
}

//??¤????????°?????????
message TechwolfGroupSync {
    // ??¤id
    required int64 gid = 1;
    // ????????·
    optional int32 version = 2;
}

//??¨???è?¨???
message TechwolfSticker {
    // è?¨???id
    required int64 sid = 1;
    // è?¨??????id
    optional int64 packId = 2;
    // è?¨?????????
    optional TechwolfImage image = 3;
    // è?¨??????????????????gif|png|bmp|jpg|jpeg|tiff|...???
    optional string format = 4;
    // è?¨???????§°
    optional string name = 5;
}

//è???¤???????
message TechwolfChatShare {
    //??????id
    required int64 shareId = 1;
    //?????????é??
    required string title = 2;
    //è???¤?è?°??????è?¨
    repeated string records = 3;
    //?????????é?¨??????
    optional string bottomText = 4;
    //è·?è?????è??URL
    optional string url = 5;
    //???é??????????¨??·
    required TechwolfUser from = 6;
    //??????????????¨??·
    required TechwolfUser to = 7;
    //è????????è???¤???¨??·
    required TechwolfUser user = 8;
}

//é??è????????
message TechwolfInterviewShare {                                    // ADD SINCE V6.13
    //é??è??id
    required int64 interviewId = 1;
    //é??è????¨??·
    required TechwolfUser user = 2;
    //?????????é??
    required string title = 3;
    //???é?¨??????
    required string bottomText = 4;
    //è·?è?????è??
    optional string url = 5;
    //é??è?????é??
    optional string interviewTime = 6;
    //é??è????°???
    optional string interviewAddress = 7;
}